### DISCUSSION TITLE
2023-08-28 정기 백엔드 개발 회의

### DISCUSSION THREAD

**Discussion by jimin52** [jiphyeonjeon-42/backend#689] [General]
안건을 자유롭게 추가해주세요

**Comment by jimin52** [comment]
## docs 상위 디렉토리로 묶기

현재 한 눈에 원하는 목차를 찾아가기 어렵다고 생각합니다.

<img width="330" alt="스크린샷 2023-08-21 오후 7 19 02" src="https://github.com/jiphyeonjeon-42/backend/assets/45284810/82f1dbbe-b16b-4f53-865d-02fa2e5e370a">

일단 목차의 순서가 원하는대로 정리되지 않습니다.
하나씩 mkdocs.yml 에 nav: 탭으로 정리할 수 있지만 그 방식은 문서를 추가할때마다 목차도 업데이트 해주어야 합니다.
그래서 상위 디렉토리로 묶어서 정리하는 방식을 생각중입니다.

<img width="306" alt="스크린샷 2023-08-21 오후 7 21 52" src="https://github.com/jiphyeonjeon-42/backend/assets/45284810/5349511d-5f4d-4f37-ba88-ab7da4637a18">

<img width="317" alt="스크린샷 2023-08-21 오후 7 21 04" src="https://github.com/jiphyeonjeon-42/backend/assets/45284810/79a4f533-3a9c-4a41-89b7-ddcc8481eadf">

대충 이런 방식을 생각하고 있는데 어떻게 분류하면 좋을지 좀 더 생각해보면 좋을 것 같습니다.
놀라운 사실은 정렬이 알파벳순이 아니라는 것입니다..💢 
mkdocs, mkdocs-material 의 navigation 정렬방식을 찾아보는 중입니다.
다른 좋은 아이디어가 있으면 추천 부탁드립니다.

**Reply by jimin52** [reply]
지금 보니 index.md 제외하고 filename 대로 정렬하는 것 같네요

**Comment by scarf005** [comment]
# RDS 개인정보 삭제

#668 에서 했던 것과 동일하게, 개인정보 보호를 위해 book과 book_info 테이블을 제외하고 DB 내 모든 정보를 삭제하면 좋을 것 같습니다.

## 로그인

다만 42 intra 로그인을 하면 여전히 개인정보가 DB에 남는데, 이 부분은 어떻게 하면 좋을지 고민이 됩니다. 

연관 링크: https://github.com/orgs/jiphyeonjeon-42/discussions/11#discussioncomment-6737128

**Comment by nyj001012** [comment]
[DOS 공격 방지 express-rate-limit 도입](#691)의 내용 적용해 보는 건 어떨까요?
express-rate-limit으로 시간 당 보낼 수 있는 요청의 횟수를 제한하는 방식입니다 :)

**Reply by scarf005** [reply]
좋은 아이디어라고 생각합니다. nginx에서 [rate limit](https://www.nginx.com/blog/rate-limiting-nginx/)을  지정하면 도서 목록 조회나 검색같이 한번에 요청을 많이 해야 하는 경로들에 예외를 주기 까다롭더라고요.

**Reply by nyj001012** [reply]
그럼 일단 #696 에서 그 알림이 오니 시범적으로 거기에 도입해봐도 될까요?

**Reply by nyj001012** [reply]
PR 테스트 통과했습니다 :) 일단 `/books/recommend`에만 적용해봤어요

**Comment by scarf005** [comment]
# 프론트엔드와 V2 API 연동

https://github.com/jiphyeonjeon-42/backend/assets/54838975/6fb78ba9-d2be-4e97-85f1-09bb4884fecc

마침내 V2를 프론트엔드와 연동하게 되었습니다.

- https://github.com/jiphyeonjeon-42/frontend/pull/475
- https://github.com/jiphyeonjeon-42/backend/pull/705

## 프론트엔드에서 `contracts` 설치법

1. `packages/`에서 `pnpm build && pnpm pack`으로 빌드를 마친 패키지를 압축 파일로 저장합니다. ([pnpm pack](https://pnpm.io/cli/pack): 패키지를 설치 가능한 tarball로 압축)
2. [github releases](https://github.com/jiphyeonjeon-42/backend/releases)에 해당 tarball을 배포합니다. ([사용한 명령어](https://cli.github.com/manual/gh_release))
3. 프론트엔드는 해당 releases에서 `@jiphyeonjeon-42/contracts` [패키지를 설치합니다](https://pnpm.io/cli/add#install-from-remote-tarball).

## 장점

- npm에 비해 패키지 배포, 수정 및 삭제 과정이 단순합니다.
- 별개로 npm organization을 관리할 필요가 없습니다.

## 고려할 점

백엔드와 프론트엔드의 `contracts` 버전이 같아야 하는데, 동기화할 방법을 찾아야 될 것 같습니다.

**Comment by jimin52** [comment]
로컬에서 Docker 안 돌아가는 이슈
